src/lib/dnssd/minimal_mdns/Server.cpp

```diff
@@ -188,6 +240,10 @@ CHIP_ERROR ServerBase::Listen(chip::Inet::InetLayer * inetLayer, ListenIterator
         {
             endpointIndex++;
         }
+
+        ReturnErrorOnFailure(inetLayer->NewUDPEndPoint(&info->query_udp));
+        ReturnErrorOnFailure(info->query_udp->Bind(addressType, chip::Inet::IPAddress::Any, kPickRandomBindPort, interfaceId));
+        ReturnErrorOnFailure(info->query_udp->Listen(OnUdpPacketReceived, nullptr /*OnReceiveError*/, this));
```

Comment:
Please add comments that explain the side-effect of using random query port: that we will always fall into RFC6762 sec 4.7, legacy unicast queries.

Maybe it would be good to have the option, depending on config, to either enable separate legacy unicast query mode, or not.

Overall, this can be determined by whether we have a socket on non-5353, or not.

Reply:
Added comments.